### PR TITLE
CPLAT-4534: Post-Dart2 cleanup

### PR DESCRIPTION
**Pull request by olesiathoms-wk**
## Motivation
  
We can finally start removing dart1 .

DO NOT MERGE waiting for dart_dev 3.0.0 release.

## Changes
  
- Updated dependencies
- function type typedefs were update to use new syntax
- unnecessary `new` were removed
- `dev.dart` was updated to use `dart_dev v3` and renamed to `config.dart`
- deprecated lint rules were removed
- this PR also fixes https://github.com/Workiva/w_flux/issues/125
-  `.pub` removed from `.gitignore`
- run `.travis.yaml` with dart 2.4.1, dev and stable versions
- add `build_vm_compilers dev_dependency`
- use `isA` instead of deprecated `isInstanceOf`

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
         - [ ] CI passes
         - [ ] run example locally and make sure that you don't see errors
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/w_flux/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/w_flux/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: tool/dart_dev/config.dart [hunk 1]
```diff
@@ -12,23 +12,11 @@
...
-    ..platforms = ['vm', 'chrome']
-    ..pubServe = true;
-
-  await dev(args);
-}
+final config = {
+  'analyze': AnalyzeTool(),
+  'format': FormatTool(),
+  'test': TestTool(),
+  'serve': WebdevServeTool()..webdevArgs = ['example:8080']
```
Comment (evanweible-wf) [inline tool/dart_dev/config.dart]: The analyze, format, and test tools are included by default in `coreConfig`:

```dart
final config = {
  ...coreConfig,
  'serve': WebdevServeTool()..webdevArgs = ['example:8080']
};
```

Comment (olesiathoms-wk) [reply tool/dart_dev/config.dart]: ok, I'll update this in `key_binder` and `w_module` as well.

File: pubspec.yaml [hunk 2]
```diff
@@ -11,23 +11,18 @@ authors:
...
-  over_react: ">=2.4.0 <4.0.0"
-  test: ">=0.12.30 <2.0.0"
-
-transformers:
-  - test/pub_serve:
-      $include: test/**_test{.*,}.dart
+  build_runner: ^1.7.1
+  build_test: ^0.10.9
+  build_web_compilers: ^2.5.1
+  dart_dev: ^3.0.0-alpha
```
Comment (olesiathoms-wk) [inline pubspec.yaml]: @evanweible-wf  what is the best way to go about `dart_dev` version?

Comment (evanweible-wf) [reply pubspec.yaml]: I'm hoping that we can get 3.0.0 stable released early next week, so let's just hold until then

**Comment by georgelesica-wf** [comment]
+1

**Comment by georgelesica-wf** [comment]
QA +1 ran the example and nothing broke @Workiva/release-management-p

**Pull request merged**